cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/CollectionComprehensionTest.kt

```diff
@@ -0,0 +1,1790 @@
+            "b",
+            refB,
+            "The first element of the tuple \"(b[a], a)\" is expected to be represented by a SubscriptExpression with in the CPG. We expect that the base is a Reference with localName \"b\"  and the subscriptExpression representing the index is a Reference \"a\" which refers to no variable available at this point.",
+        )
+        val index = accessBA.subscriptExpression
+        assertIs<Reference>(
+            index,
+            "The first element of the tuple \"(b[a], a)\" is expected to be represented by a SubscriptExpression with in the CPG. We expect that the base is a Reference with localName \"b\"  and the subscriptExpression representing the index is a Reference \"a\" which refers to no variable available at this point.",
+        )
+        assertLocalName(
+            "a",
+            index,
+            "The first element of the tuple \"(b[a], a)\" is expected to be represented by a SubscriptExpression with in the CPG. We expect that the base is a Reference with localName \"b\"  and the subscriptExpression representing the index is a Reference \"a\" which refers to no variable available at this point.",
+        )
+        val refA = tuple.initializers[1]
+        assertIs<Reference>(
+            refA,
+            "The second element of the tuple \"(b[a], a)\" is expected to be represented by a Reference with localName \"a\" in the CPG.",
+        )
+        assertLocalName(
+            "a",
+            refA,
+            "The second element of the tuple \"(b[a], a)\" is expected to be represented by a Reference with localName \"a\" in the CPG.",
+        )
+
+        // Now the actually interesting part: We check for variables belonging to the references.
+        val innerVariableDeclarationA = refA.refersTo
+        assertIs<VariableDeclaration>(
+            innerVariableDeclarationA,
+            "We expect that the reference \"a\" in the second element of the tuple refers to a VariableDeclaration with localName \"a\" which is not null and whose scope is the LocalScope of the list comprehension.",
+        )
+        assertIs<LocalScope>(
+            innerVariableDeclarationA.scope,
+            "We expect that the reference \"a\" refers to a VariableDeclaration with localName \"a\" which is not null and whose scope is the LocalScope of the list comprehension.",
+        )
+        assertEquals(
+            listComprehensionWithTupleAndAssignmentToListElement,
+            innerVariableDeclarationA.scope?.astNode,
+            "We expect that the reference \"a\" refers to a VariableDeclaration with localName \"a\" which is not null and whose scope is the LocalScope of the list comprehension.",
+        )
+        assertRefersTo(
+            refB,
+            bDeclaration,
+            "We expect that the reference \"b\" in the tuple refers to the VariableDeclaration of \"b\" which is added outside the list comprehension (in statement 0).",
+        )
+        assertRefersTo(
```

Comment:
We should discuss what we expect in this case. My version of Python3.13 crashes with the error "UnboundLocalError: cannot access local variable 'a' where it is not associated with a value".

We may interpret this by setting the `refersTo` to same as the reference `a` in the second item and later, an analysis must figure out that this variable is not initialized yet (we should check the EOG for this too).

We may also interpret this that `refersTo` should be `null`

Reply:
I would say that `refersTo` should be null. I think we did that in other cases where we have an unbound error as well

Reply:
Ok, since the `SymbolResolver` is in charge of this, I added issue #2035 which blocks this PR

Reply:
I would propose to remove the assert there, unblock and merge this PR, otherwise we will get stuck on many follow-ups, this is only a minor detail and actually has nothing to do with the PR itself.